src/lib/y2storage/dialogs/inst_disk_proposal.rb

```diff
@@ -0,0 +1,87 @@
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may
+# find current contact information at www.novell.com.
+
+require "yast"
+require "y2storage"
+require "ui/installation_dialog"
+
+Yast.import "HTML"
+
+module Y2Storage
+  module Dialogs
+    # Calculates the storage proposal during installation and provides
+    # the user a summary of the storage proposal
+    class InstDiskProposal < ::UI::InstallationDialog
+      # For Devicegraph#actiongraph
+      using Y2Storage::Refinements::Devicegraph
+
+      def initialize
+        super
+        textdomain "storage"
+      end
+
+    protected
+
+      def settings
+        settings = Y2Storage::ProposalSettings.new
+        settings.use_separate_home = true
+        return settings
+      end
+
+      def summary
+        formatted_actiongraph
+      rescue Y2Storage::Proposal::Error
+        log.error("generating proposal failed")
+        # error message
+        Yast::HTML.Para(Yast::HTML.Colorize(_("No proposal possible."), "red"))
+      end
+
+      def formatted_actiongraph
+        proposal = Y2Storage::Proposal.new(settings: settings)
+        proposal.propose
+        actiongraph = proposal.devices.actiongraph
+        texts = actiongraph.commit_actions_as_strings.to_a
+        return Yast::HTML.Para(Yast::HTML.List(texts))
```

Comment:
I just wanted to avoid a religious war by removing the `return` calls that were already there. But if I have to choose between removing them or adding more, I have no doubt. :wink: